src/types/index.ts

```diff
@@ -1,46 +1,392 @@
-// @ts-ignore
-interface AutoComplete {}
+// -----------------------------------------------------------------------------
+//  --------------------------------- TYPES ------------------------------------
+// -----------------------------------------------------------------------------
 
-interface StopGroup {}
+type GroundPlace = StopGroup | StopCluster;
 
-interface StopCluster {}
+type GroundPlaces = Record<Gpuid, GroundPlace>;
 
-interface StopGroupProperties {}
+type Gpuid = string;
 
-interface StopClusterProperties {}
+type StopGroupGpuid = Gpuid;
 
-interface Filters {}
+type StopClusterGpuid = Gpuid;
 
-interface GroundPlacesDiff {}
+// -----------------------------------------------------------------------------
+//  ------------------------------- INTERFACES ---------------------------------
+// -----------------------------------------------------------------------------
 
-type Gpuid = string;
+interface StopGroup {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.GROUP;
+  childs: SegmentProviderStop[];
+  serviced: Serviced;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopGroupGpuid = Gpuid;
+interface StopCluster {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.CLUSTER;
+  childs: StopGroupGpuid[];
+  serviced: Serviced;
+  unique_name?: string;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopClusterGpuid = Gpuid;
+interface SegmentProviderStop {
+  unique_name: null;
+  company_name: string;
+  name: string;
+  latitude: number;
+  serviced: Serviced;
+  company_id: number;
+  longitude: number;
+  id: string;
+}
+
+interface CreateStopGroupProperties {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  serviced: Serviced;
+}
+
+interface CreateStopClusterProperties {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  createdFromStopGroupGpuid: StopGroupGpuid;
```

Comment:
Not sur to get why it's in it ? 
shouldn't it be a param of the function and not a property ?

There is a micmac between types & params

Reply:
No weird concept : 
 * either this is Params either this is something that have a sense by itself